### ISSUE TITLE
Issue with 0.9.76 "Contract can't be hashed"

### ISSUE THREAD

**Issue by huangalexc** [erdewit/ib_insync#527]
This issue is specifically with 0.9.76. When I revert to 0.9.75 the error goes away.
Here is the code that causes the error:

```
amd = Stock('AMD', 'SMART', 'USD')
ib.qualifyContracts(amd)

```
Here is the traceback:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/ib.py", line 551, in qualifyContracts
    return self._run(self.qualifyContractsAsync(*contracts))
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/ib.py", line 308, in _run
    return util.run(*awaitables, timeout=self.RequestTimeout)
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/util.py", line 333, in run
    result = loop.run_until_complete(task)
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/ib.py", line 1715, in qualifyContractsAsync
    detailsLists = await asyncio.gather(
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/ib.py", line 1716, in <genexpr>
    *(self.reqContractDetailsAsync(c) for c in contracts))
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/ib.py", line 1848, in reqContractDetailsAsync
    self.client.reqContractDetails(reqId, contract)
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/client.py", line 657, in reqContractDetails
    self.send(*fields)
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/client.py", line 244, in send
    if field in empty:
  File "/Users/me/mambaforge/envs/pysystemtrade/lib/python3.9/site-packages/ib_insync/contract.py", line 153, in __hash__
    raise ValueError(f'Contract {self} can\'t be hashed')
ValueError: Contract Stock(symbol='AMD', exchange='SMART', currency='USD') can't be hashed
```

**Comment by sharpkids** [comment]
I am having the same issue with futures contracts as well:

'ValueError:  Contract Future(symbol='ES', lastTradeDateOrContractMonth='20221216', exchange='CME') can't be hashed'

**Comment by john9631** [comment]
Same here on fx contracts 

`Forex('AUDJPY', exchange='IDEALPRO') 2022-11-20 07:59:00 30 D 1 day MIDPOINT False 1 False

Secondary Exception Contract Forex('AUDJPY', exchange='IDEALPRO') can't be hashed
`
My contracts are stored in a dict, contracts in the form `'AJ': [Forex('AUDJPY'), f'{DATA}AUDJPY.scid', False, 'pb5', 15016133, None, True],`  and the line that generates the error is:  
`            l = self.ib.reqHistoricalData(self.contract, self.last_datetime + timedelta(
                days=1), depth, '1 day', self.wts, False, 1, False)`

For anyone experiencing this problem who needs to revert to 0.9.75 temporarily, the command is (adjust for your version on python if it's not 3.10):

`python3.10 -m pip install ib-insync==0.9.75`

This will show the warning re explicit time zones but that can be ignored while Ewald resolves the hashing issue with 0.9.76.

**Comment by sharpkids** [comment]
The issue is caused by this commit: https://github.com/erdewit/ib_insync/commit/71403e10d8bd67e26f3e2489df6ecd90ed4fc9c4

It's an easy fix - dict should be changed back to tuple:
empty = {None, UNSET_INTEGER, UNSET_DOUBLE}
->
empty = (None, UNSET_INTEGER, UNSET_DOUBLE)

**Comment by erdewit** [comment]
The fix from @sharpkids is available in v0.9.77.
